addons/web-support/impl/src/main/java/org/jboss/windup/web/addons/websupport/services/PackageDiscoveryServiceImpl.java

```diff
@@ -67,14 +67,24 @@ public PackageDiscoveryResult execute(String rulesPath, String inputPath)
     private void qualifyDiscoveredPackages(Map<String, String> knownPackages, PackageFrequencyTrie discoveredPackagesFrequencyTrie,
                 Map<String, Integer> knownPackagesAndClassCount, Map<String, Integer> unknownPackagesAndClassCount)
     {
+        Map<String, Integer> rootPackageUnknownClassCount = new TreeMap<String,Integer>();
+
         discoveredPackagesFrequencyTrie.visit((trie, depth) -> {
             String packageName = trie.getPackageName();
 
             int recursiveClassCount = trie.getClassCount(true);
             int nonRecursiveClassCount = trie.getClassCount(false);
 
-            Map<String, Integer> resultingMap = null;
+            boolean isRootPackage = false;
+            String rootPackageName = null;
+            String[] packageNameHierarchy = packageName.split("[.]");
+            if (packageNameHierarchy != null && packageNameHierarchy.length > 0)
+            {
+                rootPackageName = packageNameHierarchy[0];
+                isRootPackage = rootPackageName.equals(packageName);
```

Comment:
This is ok because the packageName here is the fully qualified package name of whichever package is being visited in the trie, eg 'org.apache.common', not just the subelement of it 'common'. The rootPackageName of this will be 'org'

Reply:
well, in fact it's splitting by the dot , so it's taking individual subelements, but there's no problem as it only uses the first one.